### ISSUE TITLE
Layer C vs Layer E classification dimension

### ISSUE THREAD

**Issue by katdimitris** [Yukun-Huang/Rank-Diminishing-in-Deep-Neural-Networks#1]
Hello,
As I understand it, running cls_dim.py with Layer E gives the classification dimension on the final feature manifold. Running the same script with Layer C is essentially performing the same procedure on the logits. For example in order to keep 95% original accuracy you only need 45/100 logit values for cifar100. Is this right, and if so is it connected to the independence deficit, or is it a better way to validate the general degree of independence deficit (without looking at specific classes and examples like run_deficit.py does)? 

Thank you :)

**Comment by Yukun-Huang** [comment]
Sorry for the late reply.

Yes, I think you are right. Performing dimension analysis via `cls_dim.py` on Layer C or Layer E can also reveal the independence deficit. But analyzing the relationships between different categories, as done by `run_deficit.py`, can further provide counterintuitive examples caused by the independence deficit.
